src/gen_alpha_poc/quaternion.h

```diff
@@ -157,17 +157,18 @@ Vector rotation_vector_from_quaternion(const Quaternion& quaternion) {
     auto q2 = quaternion.GetYComponent();
     auto q3 = quaternion.GetZComponent();
 
-    auto sin_angle_squared = q1 * q1 + q2 * q2 + q3 * q3;
-
-    // Return the rotation vector {0, 0, 0} if provided quaternion is null
-    if (close_to(sin_angle_squared, 0.)) {
-        return Vector{0.0, 0.0, 0.0};
+    Vector n(q1, q2, q3);                           // Vector part of quaternion
+    double n_length = n.Length();                   // magnitude of vector
+    double n_length_squared = n_length * n_length;  // magnitude squared
+
+    // If the length of the vector is less than machine precision,
+    // calculate and return rotation vector using method stable for small angles
+    if (n_length < kEpsilon) {
+        return n * (2. / q0 - 2. / 3. * n_length_squared / (q0 * q0 * q0));
```

Comment:
I might be missing it, but do we have a unit test for this behavior?

Reply:
Technically every time a quaternion has a zero vector, it will go through this branch. However, I've added a test case specifically for it.